packages/core/src/components/cms/ViewportObserver.tsx

```diff
@@ -0,0 +1,87 @@
+import type {
+  DetailedHTMLProps,
+  HTMLAttributes,
+  PropsWithChildren,
+} from 'react'
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+const viewportSize = 825 // mobile height to prevent sections that are outside the viewport from being rendered initially. We are using Moto G Power devices as a reference.
```

Comment:
suggestion: giving a bit more information and breaking line for better readability. What do you think?
```suggestion
// Mobile height to prevent sections outside the viewport from being rendered initially. 
// We are using the Moto G Power device measurement as a reference because it's the one PageSpeed Insights uses.
const VIEWPORT_SIZE = 825 
```

Reply:
Also using the uppercase on the constant name